crates/polars-plan/src/dsl/function_expr/rolling.rs

```diff
@@ -1,5 +1,185 @@
 use super::*;
 
+#[derive(Clone, PartialEq, Debug)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub enum RollingFunction {
+    Min(RollingOptions),
+    MinBy(RollingOptions),
+    Max(RollingOptions),
+    MaxBy(RollingOptions),
+    Mean(RollingOptions),
+    MeanBy(RollingOptions),
+    Sum(RollingOptions),
+    SumBy(RollingOptions),
+    Median(RollingOptions),
+    MedianBy(RollingOptions),
+    Quantile(RollingOptions),
+    QuantileBy(RollingOptions),
+    Var(RollingOptions),
+    VarBy(RollingOptions),
+    Std(RollingOptions),
+    StdBy(RollingOptions),
+    #[cfg(feature = "moment")]
+    Skew(usize, bool),
+}
+
+impl Display for RollingFunction {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        use RollingFunction::*;
+
+        let name = match self {
+            Min(_) => "rolling_min",
+            MinBy(_) => "rolling_min_by",
+            Max(_) => "rolling_max",
+            MaxBy(_) => "rolling_max_by",
+            Mean(_) => "rolling_mean",
+            MeanBy(_) => "rolling_mean_by",
+            Sum(_) => "rolling_sum",
+            SumBy(_) => "rolling_sum_by",
+            Median(_) => "rolling_median",
+            MedianBy(_) => "rolling_median_by",
+            Quantile(_) => "rolling_quantile",
+            QuantileBy(_) => "rolling_quantile_by",
+            Var(_) => "rolling_var",
+            VarBy(_) => "rolling_var_by",
+            Std(_) => "rolling_std",
+            StdBy(_) => "rolling_std_by",
+            #[cfg(feature = "moment")]
+            Skew(..) => "rolling_skew",
+        };
+
+        write!(f, "{name}")
+    }
+}
+
+impl Hash for RollingFunction {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        use RollingFunction::*;
+
+        std::mem::discriminant(self).hash(state);
+        match self {
+            #[cfg(feature = "moment")]
+            Skew(window_size, bias) => {
+                window_size.hash(state);
+                bias.hash(state)
+            },
+            _ => {},
+        }
+    }
+}
+
+macro_rules! convert {
+    ($func:ident, $s:expr, $options:expr) => {{
```

Comment:
Can we use a generic function for this?

Reply:
I tried not to use macro. If there is something wired or wrong, let me know!

https://github.com/pola-rs/polars/pull/12379/commits/ece67b12db5d63a20759064a15037efff4322419

Reply:
Thank you. Great PR. :+1: